src/ring_theory/adjoin_root.lean

```diff
@@ -101,6 +101,13 @@ variables {f}
 
 instance adjoin_root.has_coe_t : has_coe_t R (adjoin_root f) := ⟨of f⟩
 
+/-- Two `R`-`alg_hom` from `adjoin_root f` to the same `R`-algebra are the same iff
+    they agree on `root f`. -/
+@[ext] lemma alg_hom_ext [semiring S] [algebra R S] {g₁ g₂ : adjoin_root f →ₐ[R] S}
+  (h : g₁ (root f) = g₂ (root f)) : g₁ = g₂ :=
+alg_hom.coe_ring_hom_injective (ideal.quotient.ring_hom_ext
+  (polynomial.ring_hom_ext (λ r, by erw [alg_hom.commutes, alg_hom.commutes]) h))
```

Comment:
Can you add the missing `ideal.quotient.alg_hom_ext` (which should mention [`ideal.quotient.mkₐ`](https://leanprover-community.github.io/mathlib_docs/ring_theory/ideal/operations.html#ideal.quotient.mk%E2%82%90), and use that here instead? It would eliminate the `erw`.

Reply:
`ideal.quotient.alg_hom_ext` is done.